Wiredcraft/Section/Home/ViewModel/WCHomeViewModel.swift

```diff
@@ -0,0 +1,187 @@
+    public lazy var refreshAction = WCRxRefresh()
+    
+    /// page number
+    fileprivate lazy var page: Int = 1
+    
+    /// the keyword to query
+    public lazy var query: String = ""
+    
+    /// total count of all data, if have loaded data's count >= totalCount, the scrollView will show 'no more data'
+    fileprivate lazy var totalCount: Int = 0
+    
+    /// enable the scrollView refresh ability
+    /// - Parameter scrollView: target scrollView
+    public func enableRefreshScrollView(_ scrollView: UIScrollView) {
+        
+        let _ = self.refreshAction.refreshStatusBind(to: scrollView, { [weak self] in
+            /// header refresh
+            guard let `self` = self else { return }
+            self.page = 1
+            self.refreshAction.set(observable: self.usersListRequest())
+            self.refreshAction.reload.onNext(())
+            
+        }) { [weak self] in
+            /// footer refresh
+            guard let `self` = self else { return }
+            self.page += 1
+            self.refreshAction.set(observable: self.usersListRequest())
+            self.refreshAction.more.onNext(())
+        }
+
+        /// merge the refresh data to viewModel's user
+        let _ = self.refreshAction.data.subscribe(onNext: { [weak self] (response) in
+            
+            guard let `self` = self else { return }
+            /// handle error
+            self.handleResponseError(response: response as? WCUserListModel)
+            
+            /// set totalCount
+            self.totalCount = (response as? WCUserListModel)?.total_count ?? 0
+            let users = (response as? WCUserListModel)?.items ?? [WCUserModel]()
+            
+            if self.page == 1 {
+                self.users.accept(users)
+                if users.count == 0 {
+                    self.refreshAction.refreshStatus.onNext(.none)
+                }
+            } else {
+                self.users.accept(self.users.value + users)
+            }
+            // judge have more data
+            if self.users.value.count >= self.totalCount {
+                self.refreshAction.setNoMoreDataStstus()
+            }
+            
+        }).disposed(by: self.disposeBag)
+    }
+    
+    
+    /// the query users list request
+    /// - Returns: network request Observable
+    private func usersListRequest() -> Observable<WCUserListModel> {
+        return WCHomeNetworkService.usersList(query: self.query, page: self.page)
+            .observeOn(MainScheduler.instance)
+            .catchError({ (error) -> Observable<WCUserListModel> in
+                /// error happens
+                self.searchFail.onNext(error.localizedDescription)
+                return Observable<WCUserListModel>.empty()
+            })
+    }
+    
+    /// handle the error of query network
+    /// - Parameter response: the network response
+    private func handleResponseError(response: WCUserListModel?) {
+        if response == nil {
```

Comment:
Why not unwrap this with guard instead of == nil and then later force unwrap with `!`?

Reply:
I did intend to use `guard` at first,  and when i write `guard let  = response`,  i think what names are appropriate for variables,  `response` has been used,  I thought about it for a moment, but still had no idea.
so..... 
i used `if` to ignore name work